src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1256,10 +1258,16 @@ public static bool IsPossibleTupleContext(this SyntaxTree syntaxTree, SyntaxToke
             return false;
         }
 
-        public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
+        public static bool IsStartPatternContext(this SyntaxTree syntaxTree, SyntaxToken leftToken, int position)
         {
             leftToken = leftToken.GetPreviousTokenIfTouchingWord(position);
 
+            var node = leftToken.Parent;
+            while (node is ParenthesizedExpressionSyntax parenthesizedExpression)
+            {
+                node = node.Parent;
+            }
```

Comment:
doc.

Reply:
also, i feel like this will only work for 50% of cases.  i.e. when the parser has parsed things out as a parenthesized expression.  but it likely needs to be updated to work for parenthesized patterns as well.

Reply:
also... i feel like this isn't quite what you want.  First, this will be wrong if you're on the *right* side of a parenthesized expr.  i.e. `((...)) $$`.  This will get the close-paren, then walk up this tree.

What is likely better is that you handle the `(` specially below.  i.e. if you have an `(` to hte left, you then use .WalkUpParentheses just for that case, and then analyze what's to the left of the topmost `(` of the parenthesized exprs.